elements_imaging/imaging.py

```diff
@@ -152,63 +151,74 @@ class Processing(dj.Computed):
-        -> master
-        file_path: varchar(255)  # filepath relative to root data directory
-        """
-
     # Run processing only on Scan with ScanInfo inserted
     @property
     def key_source(self):
         return ProcessingTask & scan.ScanInfo
 
     def make(self, key):
-        method, task_mode = (ProcessingParamSet * ProcessingTask & key).fetch1('processing_method', 'task_mode')
+        task_mode = (ProcessingTask & key).fetch1('task_mode')
+        method, loaded_result = get_loader_result(key, ProcessingTask)
 
         if task_mode == 'load':
             if method == 'suite2p':
                 if (scan.ScanInfo & key).fetch1('nrois') > 0:
                     raise NotImplementedError(f'Suite2p ingestion error - Unable to handle ScanImage multi-ROI scanning mode yet')
-                
-                from .readers import suite2p_loader
+                loaded_suite2p = loaded_result
+                key = {**key, 'processing_time': loaded_suite2p.creation_time}
+            elif method == 'caiman':
+                loaded_caiman = loaded_result
+                key = {**key, 'processing_time': loaded_caiman.creation_time}
+            else:
+                raise NotImplementedError('Unknown method: {}'.format(method))
+        elif task_mode == 'trigger':
+            raise NotImplementedError(f'Automatic triggering of {method} analysis is not yet supported')
+        else:
+            raise ValueError(f'Unknown task mode: {task_mode}')
 
-                data_dir = pathlib.Path(get_suite2p_dir(key))
-                loaded_suite2p = suite2p_loader.Suite2p(data_dir)
-                key = {**key, 'proc_completion_time': loaded_suite2p.creation_time, 'proc_curation_time': loaded_suite2p.curation_time}
-                # Insert file(s)
-                root = pathlib.Path(scan.get_imaging_root_data_dir())
-                output_files = data_dir.glob('*')
-                output_files = [f.relative_to(root).as_posix() for f in output_files if f.is_file()]
+        self.insert1(key)
 
-            elif method == 'caiman':
-                from .readers import caiman_loader
 
-                data_dir = pathlib.Path(get_caiman_dir(key))
-                loaded_caiman = caiman_loader.CaImAn(data_dir)
+@schema
+class Curation(dj.Manual):
+    definition = """
+    -> Processing
+    curation_id: int
+    ---
+    curation_time: datetime             # time of generation of this set of curated results 
+    curation_output_dir: varchar(255)   # output directory of the curated results, relative to root data directory
+    manual_curation: bool               # has manual curation been performed on this result?
+    curation_note='': varchar(2000)  
+    """
 
-                key = {**key, 'proc_completion_time': loaded_caiman.creation_time,
-                              'proc_curation_time': loaded_caiman.curation_time}
+    def create1_from_processing_task(self, key, is_curated=False, curation_note=''):
```

Comment:
Where would we run this function?  Within `workflow-imaging/workflow_imaging/ingest.py`?  For the cases where `curation_output_dir` is distinct from `processing_output_dir`, would we create a `csv` file in `workflow-imaging/user_data` and insert that data within `workflow-imaging/workflow_imaging/ingest.py`?

Reply:
This is a convenient function provided to the users should they need it. The `Curation` table is a manual table, i.e. users will have to manually insert entries here (providing `curation_output_dir`, `curation_time`, etc.) - but they can use this `create1_from_processing_task()` function to generate/insert an entry based on the `ProcessingTask` entry, for when they want to ingest the results from the analysis suite without any curation